### PR TITLE
Add `maintainer` command to use xmu

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

File: xbstrap/__init__.py [hunk 1]
```diff
@@ -853,6 +854,28 @@ def do_prereqs(args):
                 if os.path.dirname(info.name) == "./usr/bin":
                     info.name = os.path.basename(info.name)
                     tar.extract(info, bin_dir)
+    if "xmu" in comps:
+        url = "https://api.github.com/repos/managarm/xbstrap-maintainer-utilities/tarball/master"
```
Comment (avdgrinten) [inline xbstrap/__init__.py]: We should download a release, not `master`.

File: xbstrap/__init__.py [hunk 2]
```diff
@@ -853,6 +854,28 @@ def do_prereqs(args):
...
+                    commit = info.name[-7:]
+                tar.extract(info, bin_dir)
+
+        extract_dir = os.path.join(bin_dir, f"managarm-xbstrap-maintainer-utilities-{commit}")
+        dest_dir = os.path.join(bin_dir, "xmu")
+
+        xbstrap.base.try_rmtree(dest_dir)
+        shutil.move(extract_dir, dest_dir)
+        _util.log_info("Installing xmu with npm")
+        proc = subprocess.Popen(["npm", "install"], cwd=dest_dir)
```
Comment (avdgrinten) [inline xbstrap/__init__.py:882]: It might be a good idea to check that `npm` actually exists.

File: xbstrap/__init__.py [hunk 3]
```diff
@@ -918,6 +941,28 @@ def resolve_host_paths(x):
...
+def do_maintainer(args):
+    cfg = config_for_args(args)
+    xmu_dir = os.path.join(_util.find_home(), "bin", "xmu")
+    if not os.access(os.path.join(xmu_dir, "xmu.js"), os.F_OK):
+        _util.log_err("The maintainer utilities are not installed.")
+        _util.log_info("Try running `xbstrap prereqs xmu` to install them.")
+        return
+    proc = subprocess.Popen(
+        [os.path.join(xmu_dir, "xmu.js")] + args.args, cwd=os.path.abspath(cfg.source_root)
+    )
```
Comment (avdgrinten) [inline xbstrap/__init__.py:961]: What if `node` does not exist? That's probably also worth checking for.

Comment (avdgrinten) [inline xbstrap/__init__.py:961]: Also, we simply pass through the command here. I would prefer to use the node.js tool for YAML rewriting, but not for all possible maintainer tasks.

Comment (no92) [reply xbstrap/__init__.py:961]: If we check that npm exists, this seems unnecessary here (and would be slightly more involved that a simple `shutil.which`).

**Comment by no92** [comment]
Addressed some of the comments raised above.

**Pull request merged**